src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -106,3 +109,63 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
 
     return nodejsLaunchConfig
 }
+
+/**
+ * Recursively searches directory and all nested directories until a TypeScript file is found
+ * @param dir Directory to search
+ * @returns true if at least one TypeScript file exists
+ */
+function hasTypeScriptFilesRecursive(dir:string): boolean {
+    const files = readdirSync(dir)
+
+    for (let i = 0; i < files.length; i++) {
+        const file = files[i];
+        
+        if (file.substr(-3) === '.ts'){
+            return true
+        }
+    
+        if(statSync(path.join(dir, file)).isDirectory()){
+            if(hasTypeScriptFilesRecursive(path.join(dir, file))) {
+                return true
+            }
+        }
+    }        
+    return false
+}
+
+/**
+ * For non-template debug configs (target = code), compile the project
+ * using a temporary default tsconfig.json file.
+ */
+async function compileTypeScript(config: NodejsDebugConfiguration): Promise<void> {
+    // Compile step for TS projects from source code
+    if (config.invokeTarget.target === 'code') {
+        const samBuildOutputAppRoot = path.join(config.baseBuildDir!, 'output', 'awsToolkitSamLocalResource')
```

Comment:
FYI: with the merge of the Java branch, we are no longer using `awsToolkitSamLocalResource`: https://github.com/aws/aws-toolkit-vscode/commit/8f8fd976e1c121fd2975f43c76fd216800b0bf15#diff-b36cc56a594606340ec0320f08168751338535f63e948fdcaab177666baffb79R67 ; swap this to `path.parse(config.invokeTarget.projectRoot).name`

Reply:
Thanks for the heads up.  Switched.